reps/2023-12-04-accelerated-dag.md

```diff
@@ -0,0 +1,360 @@
+|-----------------------------------------------------------------|----------------------------------------------------------------------|------------------------------------------------------------------|
+| vLLM tensor parallelism                                         |                                                                      | Reduce Ray overheads                                             |
+| vLLM pipeline parallelism                                       | P2P/cross-node GPU communication                                     | Reduce (expected) Ray overheads; Validate cross-node performance  |
+
+
+
+### Common patterns: Training workloads
+|                                                                 | Key properties / requirements                                        | Goals                                                            |
+|-----------------------------------------------------------------|----------------------------------------------------------------------|------------------------------------------------------------------|
+| Tensor-parallel distributed training (TP)                       | Iterative                                                            | Performance parity                                               |
+| Distributed data-parallel (DDP)                                 | Iterative; Collective ops that must be overlapped with backwards pass | Performance parity                                               |
+| [GPipe](https://arxiv.org/abs/1811.06965) style pipeline-parallel distributed training (PP)         | Iterative P2P GPU communication                                      | Performance parity; Increase flexibility of partitioning scheme   |
+
+
+### Advanced patterns
+|                                                                 | Key properties / requirements                                        | Goals                                                            |
+|-----------------------------------------------------------------|----------------------------------------------------------------------|------------------------------------------------------------------|
+| [Pipedream](https://arxiv.org/pdf/1806.03377.pdf) style pipeline-parallel distributed training (PP)     | Iterative P2P GPU communication                                      | Performance parity; Increase flexibility of partitioning scheme   |
+| vLLM pipeline parallelism on heterogeneous GPUs                 | Asymmetric compute                                                   | Reduce implementation burden                                     |
+| Fault-tolerant distributed serving                              | Resume execution w/o restarting everyone                             | Reduce downtime via greater recovery flexibility                 |
```

Comment:
As I understand the fault tolerance section above, failure handling will be done at the application layer by propagating the error back to the driver. In this case it seems that the entire DAG would need to be re-run. So how would the proposal help address this requirement?

It's also a little unclear to me how important this requirement is for most serving applications. Most often latencies are low (~seconds at most) and failures are rare, so needing to re-run the full inference is not a big problem.

Reply:
Restarting here meaning the worker processes, not individual tasks.